src/Domain/DomainId.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\Domain;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class DomainId implements DomainIdInterface
+{
+    private $id;
+
+    final public function __construct(string $id = null)
+    {
+        $this->id = $id;
+    }
+
+    final public function isKnown(): bool
+    {
+        return null !== $this->id;
+    }
+
+    final public function equals(DomainIdInterface $id): bool
+    {
+        return null !== $this->id && $id instanceof self && get_class($id) === get_class($this) ? $this->id === $id->id : false;
```

Comment:
Why both the instanceof and get class check?

`static::class` should also work instead of `get_class($this)`

Reply:
domain id's vary on value *and* type no? i hesitated, and still am (same goes for UUID implem btw, which might be even weirder because they should be globally unique anyway).

In general it's a safety measure until someone proves it's incorrect :) id say a subclass (other type) might alter the value/behavior.

Reply:
Seen some collisions come by with UUID on twitter 🤔 

`$id instanceof self && get_class($id) === get_class($this)`

The first check is to see whether or not they either either `self` or a child thereof. The second check narrows this down by checking whether the classes are the same or not. To me it looks like if the second check returns true, the first check would _always_ return true as well, thus is redundant. Or am I missing something?

Reply:
> Or am I missing something?

Yep SA and PHPstorm :)

With only `get_class` PHPStorm doesnt autocomplete on `$id->uuid` (private prop). And SA screams `Access to an undefined property MsgPhp\Domain\DomainIdInterface::$uuid`.

With only `instanceof` we dont vary on type.

Reply:
Fair enough, that makes sense. `static::class` instead of get class can still be done though.

Reply:
Done.